libraries/app/include/graphene/app/database_api.hpp

```diff
@@ -177,6 +177,7 @@ class database_api
        * @return the transaction at the given position
        */
       processed_transaction get_transaction( uint32_t block_num, uint32_t trx_in_block )const;
+      optional<processed_transaction> get_transaction_by_txid(transaction_id_type txid)const;
```

Comment:
Would it be better if the call returned not only the TX but also the block number that contains it (and perhaps the number of the TX in the block)?